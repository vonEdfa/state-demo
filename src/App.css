.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  color: white;
}

h1 {
  font-weight: normal;
}

.App-logo {
  height: 40vmin;
}

.App-link, a, a:visited {
  color: #09d3ac;
}
