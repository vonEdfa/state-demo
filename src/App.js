import 'date-fns';
import React from 'react';
// import logo from './logo.svg';
import Apod from './components/Apod';
import DateFnsUtils from '@date-io/date-fns';
import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
import './App.css';

function App() {

  const [selectedDate, setSelectedDate] = React.useState(Apod.defaultProps.date);

  const handleDateChange = date => {
    setSelectedDate(date);
  };

  return (
    <div className="App">
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <header className="App-header">
          {/* <img src={logo} className="App-logo" alt="logo" /> */}
          <KeyboardDatePicker
            disableToolbar
            variant="inline"
            format="yyyy-MM-dd"
            maxDate={Apod.defaultProps.date}
            margin="normal"
            id="date-picker-inline"
            label="Pick a date"
            value={selectedDate}
            onChange={handleDateChange}
            KeyboardButtonProps={{
              'aria-label': 'change date',
            }}
          />
          <Apod date={selectedDate} />
        </header>
      </MuiPickersUtilsProvider>
    </div>
  );
}

export default App;
