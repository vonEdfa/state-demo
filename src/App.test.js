import React from 'react';
import ReactDOM from 'react-dom';
import { act as domAct } from 'react-dom/test-utils';
import { act as testAct, create } from 'react-test-renderer';
import App from './App';

let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  ReactDOM.unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('App', () => {
  it('renders without errors', () => {
    const errors = [];
    try {
      ReactDOM.render(<App />, container);
    } catch (error) {
      errors.push(error);
    }
    expect(errors).toHaveLength(0);
  });

});
