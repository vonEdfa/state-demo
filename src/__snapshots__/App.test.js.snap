// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="date-picker-inline"
        id="date-picker-inline-label"
      >
        Pick a date
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
          disabled={false}
          id="date-picker-inline"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value="2019-10-31"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <button
            aria-label="change date"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      
    </div>
    <h1>
      APOD
    </h1>
    
  </header>
</div>
`;
